llm_handler.py

```
@@ -22,6 +23,249 @@ def extract_urls_from_text(text: str) -> list[str]:
+    
+    # Count Discord links in the response
+    discord_link_count = len(re.findall(r'discord\.com/channels/', response))
+    
+    # Check if web citations are present when they should be
+    web_citations_present = bool(re.search(r'\[\d+\]', response)) and web_sources
+    
+    # Determine minimum required sources (stricter for @bot mentions)
+    if is_mention_query:
+        MIN_REQUIRED_DISCORD_SOURCES = 5  # Stricter for @bot mentions
+    else:
+        MIN_REQUIRED_DISCORD_SOURCES = 3  # Normal for summary commands
```

Ruff (0.12.2)

Use ternary operator `MIN_REQUIRED_DISCORD_SOURCES = 5 if is_mention_query else 3` instead of `if`-`else`-block

(SIM108)